src/client/utils/tracking/trackOfferEvent.ts

```diff
@@ -69,3 +83,62 @@ export const trackOfferEvent = (
     captureSentryError(error)
   }
 }
+
+export const useTrackOfferEvent = () => {
+  const { isoLocale } = useCurrentLocale()
+
+  const { quoteCartId } = useQuoteCartIdFromUrl()
+
+  const [selectedInsuranceTypes] = useSelectedInsuranceTypes()
+
+  const { data: quoteCartQueryData } = useQuoteCartQuery({
+    variables: {
+      id: quoteCartId,
+      locale: isoLocale,
+    },
+  })
+
+  const isReferralCodeUsed =
+    getMonthlyCostDeductionIncentive(quoteCartQueryData) !== undefined
+
+  const selectedBundleVariant = getSelectedBundleVariant(
+    quoteCartQueryData,
+    selectedInsuranceTypes,
+  )
+
+  const [eventQueue, setEventQueue] = useState<EventParameters[]>([])
+
+  const trackOfferCallback = useCallback(
+    ({ eventName, options = {} }: EventParameters) => {
+      trackOfferEvent(
+        eventName,
+        selectedBundleVariant!.bundle,
+        isReferralCodeUsed,
+        {
+          quoteCartId,
+          ...options,
+        },
+      )
+    },
+    [isReferralCodeUsed, quoteCartId, selectedBundleVariant],
+  )
+
+  const trackOfferHandler = useCallback(
+    (eventParams: EventParameters) => {
+      selectedBundleVariant
+        ? trackOfferCallback(eventParams)
+        : setEventQueue(eventQueue.concat([eventParams]))
+    },
+    [selectedBundleVariant, eventQueue, trackOfferCallback],
+  )
```

Comment:
Can we do this instead? Then it will only do one thing - add a new event to be tracked to the queue.

It also means we get a stable reference so it won't cause `useEffect` calls to rerun.

```suggestion
  const trackOfferHandler = useCallback(
    (eventParams: EventParameters) => {
      setEventQueue((prevQueue) => [...prevQueue, eventParams])
    },
    [],
  )
```